go/vt/vttls/crl.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2021 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vttls
+
+import (
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"fmt"
+	"io/ioutil"
+	"time"
+
+	"vitess.io/vitess/go/vt/log"
+)
+
+type verifyPeerCertificateFunc func([][]byte, [][]*x509.Certificate) error
+
+func certIsRevoked(cert *x509.Certificate, crl *pkix.CertificateList) bool {
```

Comment:
Should this also validate that whoever signed `cert` here also has signed the CRL? See also https://pkg.go.dev/crypto/x509#Certificate.CheckCRLSignature.

Reply:
good idea! i can take a look at this improvement